python/thunder/utils/decorators.py

```diff
@@ -0,0 +1,281 @@
+""" Useful decorators that are used throughout the library """
+
+def _isNamedTuple(obj):
+  """Heuristic check if an object is a namedtuple."""
+  from collections import namedtuple
+  return hasattr(obj, "_fields") \
+     and hasattr(obj, "_asdict") \
+     and callable(obj._asdict)
+
+def serializable(cls):
+    '''The @serializable decorator can decorate any class to make it easy to store
+    that class in a human readable JSON format and then recall it and recover
+    the original object instance. Classes instances that are wrapped in this
+    decorator gain the serialize() method, and the class also gains a
+    deserialize() static method that can automatically "pickle" and "unpickle" a
+    wide variety of objects like so:
+
+      @serializable
+      class Visitor():
```

Comment:
Need an object here so it's `Visitor(object)`, otherwise this doesn't run as written =)